crates/exex/exex/src/context.rs

```diff
@@ -24,13 +24,13 @@ pub struct ExExContext<Node: FullNodeComponents> {
     /// Additionally, the exex can pre-emptively emit a `FinishedHeight` event to specify what
     /// blocks to receive notifications for.
     pub events: UnboundedSender<ExExEvent>,
-    /// Channel to receive [`ExExNotification`]s.
+    /// Channel to receive [`ExExNotification`](crate::ExExNotification)s.
     ///
     /// # Important
     ///
-    /// Once a an [`ExExNotification`] is sent over the channel, it is considered delivered by the
-    /// node.
-    pub notifications: Receiver<ExExNotification>,
+    /// Once an [`ExExNotification`](crate::ExExNotification) is sent over the channel, it is
+    /// considered delivered by the node.
+    pub notifications: ExExNotificationsSubscriber,
 
     /// node components
     pub components: Node,
```

Comment:
we still need backfill if we do the write ahead solution, right?

then to make this nice the ExExNotificationsSubscriber would to have access to the components so that the stream type can do backfill internally